src/stp/PlayDecider.cpp

```diff
@@ -5,12 +5,13 @@
 #include "stp/PlayDecider.hpp"
 
 namespace rtt::ai::stp {
-Play *PlayDecider::decideBestPlay(world::World *pWorld, std::vector<Play *> plays) noexcept {
-    if (lockedPlay) {
-        return lockedPlay;
+
+    Play *PlayDecider::decideBestPlay(world::World *pWorld, std::vector<Play *> plays) noexcept {
+        return interfacePlay ? interfacePlay : *std::max_element(plays.begin(), plays.end(), [&](auto &largest, auto &play) { return largest->score(pWorld) < play->score(pWorld); });
```

Comment:
This line is way too long/hard to read. Split it up in multiple lines, especially if you define a lambda function.

Reply:
also, I would suggest const auto& inside of the lambda,

Reply:
Also, you should probably check if the vector of plays is non-empty as otherwise this will be UB and will probably segfault.

Reply:
score() is non-const iirc.

The vector of `plays` is guaranteed to not be empty as decideBestPlay() is ran after initialization and in initialization we add plays to this vector.


Aggreed with the too hard to read, not sure why this is like this im fairly sure i didn't write it this way? (maybe tho idk)

Reply:
score() not being const seems like a problem? That smells of side-effects...

Reply:
But anyways, fine by me

Reply:
yeah idk man. ask jesse i guess?